### ISSUE TITLE
boolean datatype

### ISSUE THREAD

**Issue by tdierickx** [bjherger/keras-pandas#99]
**Background info:** Hello, I really like your module, but I've been wrestling with how to get a simple binary classifier to work. 

I have defined in my data type dictionary that my response variable should be categorical _(really, it's 1/0)_, like so:

```
data_type_dict= {"numerical": ["AWND", "TMAX", "TMIN", "WDF5"], "categorical": ["RESPONSE"]}
auto = Automater(data_type_dict=data_type_dict, output_var="RESPONSE")
...
model.compile(optimizer="Adam", loss="binary_crossentropy", metrics=["accuracy"])
```

The error I keep gettings is:
> ValueError: Error when checking target: expected dense_34 to have shape (3,) but got array with shape (1,)

If I change it to use loss=auto.suggest_loss() I see that it really just uses "sparse_categorical_crossentropy", it "works" without error, but you end up with 3 predicted values per observations

**Hope for a moment:** I saw within the Automater.py file, different "datatype_handlers" listed, including the "undocumented" boolean one, so I was momentarily excited that I could perhaps use that in my "data_type_dict" above but, alas, I got the same error and I now see that's because it's treated exactly like a categorical field with multiple values, per next comment...

**Root issue:** I see inside the output_nub_generator() function in "data_types/Categorical.py", the following lines:

```
# +1 for UNK level
categorical_num_response_levels = len(set(transformed_observations[variable])) + 1
output_layer = Dense(units=categorical_num_response_levels, activation='softmax')
return output_layer
```
So, this explains why I kept having problems today: it's implicitly adding a 3rd level to support missing values _(which, *could* be great, except in my case I have all 1/0 without any NaN's)_ and it's hard-coded to use "softmax" instead of "sigmoid". 

**Possible workaround:** I will just look to use the predicted probabilities for the first two levels and ignore the 3rd one; e.g., maybe something like so: P(1)* = P(1) / [P(1) + P(2)] and ignore P(3)

**Question:** is this how you would advise people to do it? Or is there future consideration to an extra parameter, or list item, to specify whether certain columns are expected to be complete _(no NaN)_ or not?

**Comment by tdierickx** [comment]
**Follow-up:** it looks I can also just pile on an extra "final" last layer like so to then be able to use other loss functions _(i.e. "binary_crossentropy" in my case)_:

```
x = auto.input_nub
...
x = auto.output_nub(x)
x = Dense(1, activation="sigmoid")(x)
...
model = Model(inputs=auto.input_layers, outputs=x)
model.compile(optimizer='adam', loss="binary_crossentropy")
```
The resulting model.predict(X) predicted probabilities look fine

_note: actually better than before where was "forced" to use "sparse_categorical_crossentropy", whether explicitly setting or implicitly setting via auto.suggest_loss() suggested_ 

However, the auto.inverse_transform_output(predicted_probs) now appear "wrong" _(i.e. predicted prob=0.522 => has predicted value=0 when should be 1, since >= 0.500)_ and I think your documentation implies this likely would be the case

**Addendum:** 
it looks like _*any*_ of the Keras loss functions that involve axis=-1 fail if try just going with **model.compile(..., outputs=auto.output_nub(x) )** by itself; the list as I see them are here: https://github.com/keras-team/keras/blob/master/keras/losses.py and, so, only "categorical_crossentropy" and "sparse_categorical_crossentropy" seem to work here when using auto.output_nub(x)

**Comment by bjherger** [comment]
This is a great catch, thanks for bringing it up. 

**Current state**
Previously, I'd treated bools and categoricals as the same, to reduce code spaghetti. You're right in stating that the bool and categorical behavior is similar for input variables, but very different for output behavior. 

**Future state**
keras-pandas should have a separate `Boolean` datatype. Because implementing datatypes has gotten a lot easier with the v3.X datatype format, this should be a quick fix.

**Comment by bjherger** [comment]
Will attempt to tackle this in the upcoming Friday release

**Comment by bjherger** [comment]
Addressed in #104 

@tdierickx
